File: adminpanel/update_buttons.php
<?php
$files = [
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/shipping/rules-table.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/shipping/zone-table.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/shipping/state-table.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/notification/notification-list.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/contact/contact-enquire.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/admin/roles/role-list.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/admin/users/user-list.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/admin/users/list.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/admin/roles/list.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/contact/contact-enquiries.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/notification/table.blade.php',
  'c:/xampp/htdocs/chennais/adminpanel/resources/views/offer/offer-table.blade.php'
];

foreach (glob('c:/xampp/htdocs/chennais/adminpanel/resources/views/*/*.blade.php') as $f) { $files[] = $f; }
foreach (glob('c:/xampp/htdocs/chennais/adminpanel/resources/views/*/*/*.blade.php') as $f) { $files[] = $f; }

foreach (array_unique($files) as $f) {
  if (!file_exists($f)) continue;

  $content = file_get_contents($f);

  $content = preg_replace(
    '/<a((?:.|\n)*?)class="btn((?:.|\n)*?)btn-warning((?!btn-action-col)(?:.|\n)*?)"((?:.|\n)*?)>\s*<i\s+class="bi bi-pencil(?:-square)?(\s+me-1)?"><\/i>\s*<\/a>/i',
    '<a$1class="btn$2btn-warning btn-action-col d-inline-flex align-items-center justify-content-center$3"$4><i class="bi bi-pencil-square me-1"></i> Edit</a>',
    $content
  );
  if(preg_match('/class="btn[^"]*btn-warning[^"]*"[^>]*>\s*<i[^>]*bi-pencil[^>]*><\/i>\s*Edit\s*<\/a>/i', $content)){
      $content = preg_replace(
        '/<a((?:.|\n)*?)class="btn((?:.|\n)*?)btn-warning((?!btn-action-col)(?:.|\n)*?)"((?:.|\n)*?)>\s*<i\s+class="bi bi-pencil(?:-square)?(\s+me-1)?"><\/i>\s*Edit(\s*)<\/a>/i',
        '<a$1class="btn$2btn-warning btn-action-col d-inline-flex align-items-center justify-content-center$3"$4><i class="bi bi-pencil-square me-1"></i> Edit</a>',
        $content
      );
  }

  $content = preg_replace(
    '/<button((?:.|\n)*?)class="btn((?:.|\n)*?)btn-danger((?!btn-action-col)(?:.|\n)*?)"((?:.|\n)*?)>\s*<i\s+class="bi bi-trash(\s+me-1)?"><\/i>\s*<\/button>/i',
    '<button$1class="btn$2btn-danger btn-action-col d-inline-flex align-items-center justify-content-center$3"$4><i class="bi bi-trash me-1"></i> Delete</button>',
    $content
  );
  
  if(preg_match('/class="btn[^"]*btn-danger[^"]*"[^>]*>\s*<i[^>]*bi-trash[^>]*><\/i>\s*Delete\s*<\/button>/i', $content)){
      $content = preg_replace(
        '/<button((?:.|\n)*?)class="btn((?:.|\n)*?)btn-danger((?!btn-action-col)(?:.|\n)*?)"((?:.|\n)*?)>\s*<i\s+class="bi bi-trash(\s+me-1)?"><\/i>\s*Delete\s*<\/button>/i',
        '<button$1class="btn$2btn-danger btn-action-col d-inline-flex align-items-center justify-content-center$3"$4><i class="bi bi-trash me-1"></i> Delete</button>',
        $content
      );
  }
  
  $content = preg_replace(
    '/<a((?:.|\n)*?)class="btn((?:.|\n)*?)btn-info((?!btn-action-col)(?:.|\n)*?)"((?:.|\n)*?)>\s*<i\s+class="bi bi-eye(\s+me-1)?"><\/i>\s*<\/a>/i',
    '<a$1class="btn$2btn-info btn-action-col d-inline-flex align-items-center justify-content-center$3"$4><i class="bi bi-eye me-1"></i> View</a>',
    $content
  );
  if(preg_match('/class="btn[^"]*btn-info[^"]*"[^>]*>\s*<i[^>]*bi-eye[^>]*><\/i>\s*View\s*<\/a>/i', $content)){
      $content = preg_replace(
        '/<a((?:.|\n)*?)class="btn((?:.|\n)*?)btn-info((?!btn-action-col)(?:.|\n)*?)"((?:.|\n)*?)>\s*<i\s+class="bi bi-eye(\s+me-1)?"><\/i>\s*View\s*<\/a>/i',
        '<a$1class="btn$2btn-info btn-action-col d-inline-flex align-items-center justify-content-center$3"$4><i class="bi bi-eye me-1"></i> View</a>',
        $content
      );
  }

  file_put_contents($f, $content);
}
?>

File: adminpanel/update_buttons.py
import os
import re
import glob

def process_file(filepath):
    try:
        with open(filepath, "r", encoding="utf-8") as f:
            content = f.read()

        # Inject JavaScript into app.blade.php
        if "app.blade.php" in filepath and "function updateTableBadges()" not in content:
            script = """
    <script>
        document.addEventListener("DOMContentLoaded", function() {
            function unifyTableBadges() {
                var $tables = $('.table');
                $tables.each(function() {
                    var $table = $(this);
                    
                    if ($table.attr("data-badges-unified") === "true") return;
                    
                    var $headerRow = $table.find("thead tr, .table-light tr");
                    if ($headerRow.length === 0) return;
                    
                    var $headers = $headerRow.find("th");
                    $headers.each(function(colIndex) {
                        var headerText = $(this).text().trim().toLowerCase();
                        if (headerText === "status") {
                            var $elements = $table.find("tbody tr").map(function() {
                                return $(this).children("td").eq(colIndex).find(".badge, .btn-sm").get();
                            });
                            
                            if ($elements.length > 0) {
                                var maxWidth = 0;
                                var targetRadius = "4px";
                                var hasPill = false;
                                
                                $elements.each(function() {
                                    var $el = $(this);
                                    $el.css({ "width": "auto", "min-width": "0" });
                                    var w = $el.outerWidth();
                                    if ($el.hasClass("rounded-pill")) {
                                        hasPill = true;
                                    }
                                    if (w > maxWidth) {
                                        maxWidth = w;
                                        if (!hasPill) {
                                            targetRadius = $el.css("border-radius");
                                        }
                                    }
                                });
                                
                                if (hasPill) {
                                    targetRadius = "50rem";
                                }
                                
                                if (maxWidth > 0) {
                                    $elements.each(function() {
                                        $(this).css({
                                            "width": (maxWidth + 10) + "px", 
                                            "display": "inline-block",
                                            "text-align": "center",
                                            "border-radius": targetRadius
                                        });
                                    });
                                }
                            }
                        }
                    });
                    
                    $table.attr("data-badges-unified", "true");
                });
            }
            
            unifyTableBadges();
            
            $(document).ajaxComplete(function(event, xhr, settings) {
                setTimeout(function() {
                    $('.table').attr("data-badges-unified", null);
                    unifyTableBadges();
                }, 100);
            });
        });
    </script>
</body>"""
            content = content.replace("</body>", script)

        # Regex modifications for action buttons
        # 1. Edit Button (icon to Edit text)
        content = re.sub(
            r"(?i)<a([^>]*?)class=[\"']btn([^>]*?)btn-warning((?!btn-action-col)[^>]*?)[\"']([^>]*?)>\s*<i\s+class=[\"']bi bi-pencil(?:-square)?(\s+me-1)?[\"']><\/i>(?:\s*Edit\s*)?<\/a>",
            r'<a\1class="btn\2btn-warning btn-action-col d-inline-flex align-items-center justify-content-center\3"\4><i class="bi bi-pencil-square me-1"></i> Edit</a>',
            content
        )
        
        # 2. Delete Button
        content = re.sub(
            r"(?i)<button([^>]*?)class=[\"'](?:btn([^>]*?)btn-danger|btn-danger((?!btn-action-col)[^>]*?))(?!btn-action-col)([^>]*?)[\"']([^>]*?)>\s*<i\s+class=[\"']bi bi-trash(\s+me-1)?[\"']><\/i>(?:\s*Delete\s*)?<\/button>",
            r'<button\1class="btn btn-sm btn-danger btn-action-col d-inline-flex align-items-center justify-content-center"\5><i class="bi bi-trash me-1"></i> Delete</button>',
            content
        )
        
        # 3. Info / View button
        content = re.sub(
            r"(?i)<button([^>]*?)class=[\"']btn([^>]*?)btn-info((?!btn-action-col)[^>]*?)[\"']([^>]*?)>\s*<i\s+class=[\"'](?:bi bi-speedometer2|bi bi-eye)(\s+me-1)?[\"']><\/i>(?:\s*Units|\s*View)?<\/button>",
            r'<button\1class="btn\2btn-info btn-action-col d-inline-flex align-items-center justify-content-center\3"\4><i class="bi bi-eye me-1"></i> View</button>',
            content
        )
        content = re.sub(
            r"(?i)<a([^>]*?)class=[\"']btn([^>]*?)btn-info((?!btn-action-col)[^>]*?)[\"']([^>]*?)>\s*<i\s+class=[\"'](?:bi bi-speedometer2|bi bi-eye)(\s+me-1)?[\"']><\/i>(?:\s*Units|\s*View)?<\/a>",
            r'<a\1class="btn\2btn-info btn-action-col d-inline-flex align-items-center justify-content-center\3"\4><i class="bi bi-eye me-1"></i> View</a>',
            content
        )
        
        with open(filepath, "w", encoding="utf-8") as f:
            f.write(content)
            
    except Exception as e:
        print(f"Error processing {filepath}: {e}")

# Process app.blade.php
process_file(r"c:\xampp\htdocs\chennais\adminpanel\resources\views\layouts\app.blade.php")

# Traverse all view templates
for root, dirs, files in os.walk(r"c:\xampp\htdocs\chennais\adminpanel\resources\views"):
    for file in files:
        if file.endswith(".blade.php"):
            process_file(os.path.join(root, file))

print("All blade templates safely processed.")
